packages/shared/src/components/auth/AuthSignBack.tsx

```diff
@@ -1,72 +1,79 @@
-import React, { ReactElement, ReactNode, useMemo } from 'react';
-import { storageWrapper as storage } from '../../lib/storageWrapper';
+import React, { MouseEventHandler, ReactElement, ReactNode } from 'react';
 import AuthHeader from './AuthHeader';
-import { AuthFormProps, Provider, providerMap } from './common';
-import OrDivider from './OrDivider';
-import ProviderButton from './ProviderButton';
+import { AuthFormProps, providerMap } from './common';
 import AuthModalFooter from './AuthModalFooter';
 import AuthContainer from './AuthContainer';
-import { ButtonSize } from '../buttons/Button';
+import { useSignBack } from '../../hooks/auth/useSignBack';
+import { ProfilePicture } from '../ProfilePicture';
+import { SignBackButton } from './SignBackButton';
+import LoginForm, { LoginFormProps } from './LoginForm';
+import ConditionalWrapper from '../ConditionalWrapper';
 
 interface AuthSignBackProps extends AuthFormProps {
   children?: ReactNode;
   onRegister?: () => void;
   onProviderClick?: (provider: string) => unknown;
+  loginFormProps?: LoginFormProps;
+  onShowLoginOptions?: MouseEventHandler;
 }
 
-const providers = Object.values(providerMap);
-
-export const SIGNIN_METHOD_KEY = 'signin_method';
-
 export const AuthSignBack = ({
-  children,
+  loginFormProps,
   onRegister,
   onProviderClick,
   simplified,
+  onShowLoginOptions,
 }: AuthSignBackProps): ReactElement => {
-  const signback = useMemo<Provider>(() => {
-    const method = storage.getItem(SIGNIN_METHOD_KEY);
-    storage.removeItem(SIGNIN_METHOD_KEY);
-    const provider = providerMap[method];
-    return provider || providerMap.google;
-  }, []);
+  const { signBack, provider, isLoaded } = useSignBack();
+  const providerItem = providerMap[provider.toLowerCase()];
+
+  if (!isLoaded || !signBack || (provider !== 'password' && !providerItem)) {
+    return null;
+  }
 
   return (
     <span className="flex flex-col flex-1">
-      <AuthHeader simplified={simplified} title="Login to daily.dev" />
-      <AuthContainer>
-        <p className="mb-2 text-center typo-callout text-theme-label-tertiary">
-          Sign back in with
+      <AuthHeader simplified={simplified} title="Welcome back!" />
+      <AuthContainer className="items-center">
+        <p className="mb-2 text-center typo-callout text-theme-label-secondary">
+          Log in to access your account
         </p>
-        <ProviderButton
-          provider={signback.provider}
-          label="Login with"
-          onClick={() => onProviderClick?.(signback.provider)}
-          {...signback}
+        <ConditionalWrapper
```

Comment:
Wait this shouldn't be conditionally wrapped right, rather conditionally rendered?
We should not show this part if you are on email login flow?

Reply:
We need the profile picture for both ends. It will need a wrapper conditionally to display the social provider icon.

Reply:
Oh I see, ok that makes sense, but does it not render completely different?
Like one is in a button, and one is just the big version of profile image?

Reply:
No, just really needs the wrapper with a relative position for the absolute placement of the icon.

Reply:
Ah ok, closing this as resolved then.